tap_dashboard_reports/streams.py

```diff
@@ -7,11 +7,13 @@
 from tap_dashboard_reports.client import Client
 from tap_dashboard_reports.period import get_iterator
 
+
 class ReportStream(Stream):
     def __init__(self, tap=None, report=None):
         self._report = report
         self._query_template = self._load_query_template()
         self._query_params = self._fetch_query_params()
+        self.custom_key = self._report.get("key_property", None) is not None
```

Comment:
self.custom_key = self._report.get("key_property", None)

If self.custom_key is not None:
   ...
   self.custom_key